core/shared/src/main/scala/zio/FiberRefs.scala

```diff
@@ -77,65 +77,62 @@ final class FiberRefs private (
    * preservation of maximum information from both child and parent refs.
    */
   def joinAs(fiberId: FiberId.Runtime)(that: FiberRefs): FiberRefs = {
+    @tailrec
+    def findAncestor(ref: FiberRef[Any])(
+      parentStack: List[(FiberId.Runtime, Any, Int)],
+      parentDepth: Int,
+      childStack: List[(FiberId.Runtime, Any, Int)],
+      childDepth: Int
+    ): Any =
+      (parentStack, childStack) match {
```

Comment:
Can you rewrite the tuple creation / matching as nested pattern match? 🙏

Reply:
Implemented